File: 3-Semestre-PUC/AEDS-II/Tps/tp01/Somador.java
import java.util.Scanner;

class Somador {

    public static int somador(int numero) {
        int resultado = 0;
        if (numero > 0) {
            resultado = (numero % 10) + somador(numero / 10);
        }
        return resultado;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (sc.hasNextInt()) {
            int numero = sc.nextInt();
            System.out.println(somador(numero));
        }
        sc.close();
    }
}

File: 3-Semestre-PUC/AEDS-II/Testes/somador.c
#include <stdio.h>
#include <stdlib.h>


int somador(int numero){
    int resultado = 0;
    while(numero > 0){
        resultado = resultado + (numero % 10);
        numero = numero / 10;
    }
    return resultado;
}


int main(){
    int numero;
    
    while(scanf("%d", &numero) == 1 && numero != 0){
        int resultado = somador(numero);
        printf("%d\n", resultado);
    }
    
    return 0;
}
